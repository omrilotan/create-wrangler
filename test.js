import { promises } from "node:fs";
import { createWrangler } from "./index.js";

const output = "fixtures/wrangler.toml";

describe("create-wrangler", () => {
	afterEach(async () => {
		await promises.rm(output);
	});
	it("should create a wrangler.toml file from a template", async () => {
		const input = "fixtures/template.1.js";
		await createWrangler({ input, output });
		const context = await promises.readFile(output, "utf8");
		expect(context).toMatchSnapshot();
	});
});
