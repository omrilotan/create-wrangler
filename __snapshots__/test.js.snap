// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`create-wrangler should create a wrangler.toml file from a template 1`] = `
"name = 'my-worker'
main = 'src/index.ts'
compatibility_date = '2022-07-04'
workers_dev = false
account_id = '96b57b07c3394752a79b623f49bc5163'

[env.production]
name = 'my-worker-production'
main = 'src/index.ts'
routes = [
	{ pattern = 'example.com/', zone_name = 'example.com' },
	{ pattern = 'example.com/app*', zone_name = 'example.com' },
	{ pattern = 'example.com/admin*', zone_name = 'example.com' },
	{ pattern = 'www.example.com/', zone_name = 'example.com' },
	{ pattern = 'www.example.com/app*', zone_name = 'example.com' },
	{ pattern = 'www.example.com/admin*', zone_name = 'example.com' },
	{ pattern = 'es.example.com/', zone_name = 'example.com' },
	{ pattern = 'es.example.com/app*', zone_name = 'example.com' },
	{ pattern = 'es.example.com/admin*', zone_name = 'example.com' },
]
kv_namespaces = [
	{ binding = 'MY_KV_NAMESPACE', id = '798d14f4b4c949d185fcf39b0f593ac7' },
]

[env.production.vars]
LOG_LEVEL = 'warn'

[[env.production.unsafe.bindings]]
type = 'analytics_engine'
name = 'MY_WORKER_ANALYTICS'
dataset = 'MY_WORKER_ANALYTICS'"
`;
